hpcflow/archive/cloud/cloud.py

```diff
@@ -1,45 +1,48 @@
-"""`hpcflow.archive.cloud.cloud.py
+"""`hpcflow.archive.cloud.cloud.py"""
+import fnmatch
+from abc import ABC
+from pathlib import Path
+from typing import List
 
-TODO: if we have multiple cloud providers, perhaps it would be better to
-implement the `upload_directory` function generally, and then just rely on 
-the provider-specific `upload_file` functions.
+from hpcflow.archive.errors import ArchiveError
 
-"""
 
-import enum
+class CloudProvider(ABC):
+    """A cloud provider is a service that allows the archival of files to a remote location."""
+    def check_access(self) -> bool:
```

Comment:
Instead of explicitly raising `NotImplementedError`, we can decorate these ABC methods with `abc.abstractmethod`.